pkgcore/ebuild/atom.py

```diff
@@ -11,6 +11,7 @@
 __all__ = ("atom", "transitive_use_atom")
 
 import string
+from itertools import product
```

Comment:
On Fri, Jul 29, 2016 at 4:08 PM, Denis Romanchuk <notifications@github.com>
wrote:

> In pkgcore/ebuild/atom.py
> <https://github.com/pkgcore/pkgcore/pull/217#discussion_r72873036>:
>
> > @@ -11,6 +11,7 @@
> >  __all__ = ("atom", "transitive_use_atom")
> >
> >  import string
> > +from itertools import product
>
> Thanks =) Doesn't pmerge always redirect all the work to the
> transitive_use_atom.evaluate_conditionals ? which then just calculate
> conditional use dependency flags based on use flags provided by a parent.
> Or have I missed something I suspect might be related to 2 ?
> I only see usage of that functionality in two places:
> 1. pquery while printing the revdeps of the package and intersecting with
> generated atoms(with all possible permutations of conditional use flags)
> 2. requesting for attribute change in package wrapper and then attempt
> force_{True,False} essentially trying to make corresponding restriction
> True or False.
> Could you provide some examples where the second case might be
> applied/needed ? like forcing some attribute to change it's state despite
> having some initial configuration.
>

I don't have anything on hand, and these days, anything I comment, it's
best to assume I'm working from past knowledge.  I'm heavily fallible, in
other words. :)

What you're describing does sound about right- that it generated the
intermediate expanded conditional form, then evaluated it to calculate down
to a locked form of the dependencies.  Assuming that's the case, then the
only issue would be the 2^25 item walk- OOM shouldn't be possible there.

I'd still double check that I didn't add some caching in for this, but I
think you're likely right.


> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/pkgcore/pkgcore/pull/217/files/469ffd8402456f4c3a2b853c521ce924393359e3#r72873036>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABddLDsAR4uKDI9ZkssT3nvJSrRRj7Jdks5qaofmgaJpZM4JXtV2>
> .
>